tests/e2e/standalone/list_test.go

```diff
@@ -0,0 +1,152 @@
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package standalone_test
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestStandaloneList(t *testing.T) {
+	ensureDaprInstallation(t)
+
+	executeAgainstRunningDapr(t, func() {
+		output, err := cmdList("")
+		t.Log(output)
+		require.NoError(t, err, "dapr list failed")
+		listOutputCheck(t, output, true)
+
+		output, err = cmdList("table")
+		t.Log(output)
+		require.NoError(t, err, "dapr list failed")
+		listOutputCheck(t, output, true)
+
+		output, err = cmdList("json")
+		t.Log(output)
+		require.NoError(t, err, "dapr list failed")
+		listJsonOutputCheck(t, output)
+
+		output, err = cmdList("yaml")
+		t.Log(output)
+		require.NoError(t, err, "dapr list failed")
+		listYamlOutputCheck(t, output)
+
+		output, err = cmdList("invalid")
+		t.Log(output)
+		require.Error(t, err, "dapr list should fail with an invalid output format")
+
+		// We can call stop so as not to wait for the app to time out
+		output, err = cmdStop("dapr_e2e_list")
+		t.Log(output)
+		require.NoError(t, err, "dapr stop failed")
+		assert.Contains(t, output, "app stopped successfully: dapr_e2e_list")
+	}, "run", "--app-id", "dapr_e2e_list", "-H", "3555", "-G", "4555", "--", "bash", "-c", "sleep 10 ; exit 0")
+
+	t.Run("daprd instance in list", func(t *testing.T) {
+		homeDir, err := os.UserHomeDir()
+		require.NoError(t, err)
+
+		path := filepath.Join(homeDir, ".dapr")
+		binPath := filepath.Join(path, "bin")
+		daprdPath := filepath.Join(binPath, "daprd")
+
+		if runtime.GOOS == "windows" {
+			daprdPath += ".exe"
+		}
+
+		cmd := exec.Command(daprdPath, "--app-id", "daprd_e2e_list", "--dapr-http-port", "3555", "--dapr-grpc-port", "4555", "--app-port", "0")
+		cmd.Start()
+
+		output, err := cmdList("")
+		t.Log(output)
+		require.NoError(t, err, "dapr list failed with daprd instance")
+		listOutputCheck(t, output, false)
+
+		// TODO: remove this condition when `dapr stop` starts working for Windows.
+		// See https://github.com/dapr/cli/issues/1034.
+		if runtime.GOOS != "windows" {
+			output, err = cmdStop("daprd_e2e_list")
+			t.Log(output)
+			require.NoError(t, err, "dapr stop failed")
+			assert.Contains(t, output, "app stopped successfully: daprd_e2e_list")
+		}
+
+		cmd.Process.Kill()
+	})
+
+	t.Run("dashboard instance should not be listed", func(t *testing.T) {
+		if isSlimMode() {
+			t.Skip("skipping test in slim mode")
+		}
+		go cmdDashboard("5555")
```

Comment:
Do we need to ensure that dashboard is launched before we execute the cmdList?

Reply:
Thinking more about it - `dapr dashboard` is a blocking call, so the spawn function will be hanged. We will need a special handling to ensure that this is working.

Reply:
Why is this being skipped in slim init?

Reply:
Removed it, thanks for the catch! Have also updated the handling mechanism, but it ends up creating a background `process` since there is no elegant way to kill the dashboard process. Let me know your thoughts.

Reply:
And this is also currently causing the Uninstall test to fail, since the dashboard binary can't be deleted.

Reply:
FYI looks like Dapr was uninstalled in the previous versions of the test,
so dashboard command would always fail and this test would pass.

https://github.com/dapr/cli/blob/c14bd7b7c05a778ce4c16ff2e261c7270e4f4bdd/tests/e2e/standalone/standalone_test.go#L152-L157

Reply:
As discussed offline, removing this test from Windows now due to its inability to delete .bin when dashboard is running.